app/controllers/users_controller.rb

```diff
@@ -0,0 +1,67 @@
+
+class UsersController < ApplicationController
+  skip_before_action :require_login,only: [:feed]
+
+  def show
+    @creators = current_user.creators
+  end
+
+  def feed
+    #user not logged in
+   if session[:user_id].nil?
+     redirect_to login_path
+   else
+     # check for new content from all the creators
+     current_user.creators.each do |creator|
+      # update creator from api if last updated more than 6 hours ago
+      if creator.last_updated.nil? || creator.last_updated < 6.hours.ago
+        creator.get_content
+      end
+     end
+     @contents = current_user.content.flatten
+     @contents = @contents.flatten.sort_by! { |content| content[:create_time] }.reverse
+     @contents = @contents.paginate(:page => params[:page], :per_page => 10)
+     render "feed"
+   end
```

Comment:
This conditional in the `feed` action is not necessary if you don't skip the `:require_login` before_action.   If the user is not logged in the before_action does the same thing as line 12.